### PR TITLE
Fix MSVC compiler and Visual Studio warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**
* reg.: declaration of 'bendFor' hides previous local declaration (C4456)
* reg.: declaration of 'groupBox' hides class member (C4458)
* reg.: Function definition for 'xMostEdgeAtY' not found (VCR001) and remove a superfluous include
* reg. 'argument': conversion from 'size_t' to 'int', possible loss of data (C4267)

### CODE REVIEW

File: src/engraving/compat/midi/compatmidirenderinternal.cpp [hunk 1]
```diff
@@ -750,7 +750,6 @@ static void collectNote(EventsHolder& events, const Note* note, const CollectNot
                             note).ticks(), pitchWheelRenderer, noteEffect);
         }
     } else {
-        const GuitarBend* bendFor = note->bendFor();
         if (bendFor) {
```
Comment (cbjeukendrup) [inline src/engraving/compat/midi/compatmidirenderinternal.cpp:753]: This can now become and `else if`

File: src/engraving/dom/trill.cpp [hunk 2]
```diff
@@ -295,7 +295,7 @@ PointF Trill::trillLinePos(const SLine* line, Grip grip, System** system)
 
     // Stop line before grace notes
     if (graceNoteSeg) {
-        const EngravingItem* preAppendedItem = graceNoteSeg->preAppendedItem(line->track2());
+        const EngravingItem* preAppendedItem = graceNoteSeg->preAppendedItem(static_cast<int>(line->track2()));
```
Comment (cbjeukendrup) [inline src/engraving/dom/trill.cpp:298]: I'd prefer changing the signatures of `preAppendedItem` and related methods to take track_idx_t

Comment (Jojo-Schmitz) [reply src/engraving/dom/trill.cpp:298]: OK, will do. It'll be quite a bit larger

**Review by cbjeukendrup — APPROVED** [review]
Thanks!

**Pull request merged**